File: 05.String字符串/leetcode_1784_checkOnesSegment.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        s += '0'
        return s.count('10') <= 1

if __name__ == "__main__":
    sl = Solution()
    s = "1001"
    print(sl.checkOnesSegment(s))

File: 05.String字符串/leetcode_1784_checkOnesSegment.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    bool checkOnesSegment(string s) {
        s += '0';
        int res = 0;
        for (int i=0; i<s.size()-1; i++){
            if (s[i] == '1' && s[i+1] == '0'){
                res++;
            }
        }
        return res<=1;
    }
};

int main(){
    Solution sl;
    string s = "1001";
    cout << sl.checkOnesSegment(s) << endl;
}
